sqUnixDragDrop.c missing a declaration of fileRecordSize()

sqUnixDragDrop.c fails to build due to a missing declaration of fileRecordSize()

Error log
```
clang -Wall  -g -O2 -DNDEBUG -DDEBUGVM=0 -msse2 -DCOGMTVM=0 -pthread -DLSB_FIRST=1 -m64 -Wno-missing-braces -Wno-unknown-pragmas -Wno-unused-value -Wno-unused-label -Wno-unused-function -Wno-unused-variable -Wno-unused-but-set-variable -I/usr/include -I/usr/include  -DHAVE_CONFIG_H  -DSQUEAK_BUILTIN_PLUGIN -I/home/faith/tmp/build/opensmalltalk-vm/building/linux64x64/squeak.cog.spur/build -I/home/faith/tmp/build/opensmalltalk-vm/building/linux64x64/squeak.cog.spur/build -I/home/faith/tmp/build/opensmalltalk-vm/platforms/unix/vm -I/home/faith/tmp/build/opensmalltalk-vm/platforms/Cross/vm -I/home/faith/tmp/build/opensmalltalk-vm/src/spur64.cog -I/usr/local/include -I/home/faith/tmp/build/opensmalltalk-vm/platforms/Cross/plugins/DropPlugin  -m64 -Wno-missing-braces -Wno-unknown-pragmas -Wno-unused-value -Wno-unused-label -Wno-unused-function -Wno-unused-variable -Wno-unused-but-set-variable -I/usr/include -I/usr/include -I/home/faith/tmp/build/opensmalltalk-vm/platforms/Cross/plugins/FilePlugin  -c -o sqUnixDragDrop.o /home/faith/tmp/build/opensmalltalk-vm/platforms/unix/plugins/DropPlugin/sqUnixDragDrop.c
/home/faith/tmp/build/opensmalltalk-vm/platforms/unix/plugins/DropPlugin/sqUnixDragDrop.c:120:66: error: call to undeclared function 'fileRecordSize'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
  120 |                 sqInt handle = instantiateClassindexableSize(classByteArray(), fileRecordSize());
      |                                                                                ^
1 error generated.
```

--- a/platforms/RiscOS/plugins/SoundPlugin/sqRPCSound.c
+++ b/platforms/RiscOS/plugins/SoundPlugin/sqRPCSound.c
@@ -285,7 +285,7 @@ sqInt snd_InsertSamplesFromLeadTime(sqInt frameCount, void *srcBufPtr, sqInt sam
 	return 0;
 }
 
-sqInt snd_PlaySamplesFromAtLength(sqInt frameCount, void *arrayIndex, sqInt startIndex) {
+sqInt snd_PlaySamplesFromAtLength(sqInt frameCount, void *buf, sqInt startIndex) {
 // Output a frameCount sound samplesfrom the passed in array, starting from startIndex.
 // return the number actually 'played'; though no one uses it
 	if (!sound_initialised) {
@@ -293,8 +293,8 @@ sqInt snd_PlaySamplesFromAtLength(sqInt frameCount, void *arrayIndex, sqInt star
 		PRINTF(("snd_PlatSamplesFrom failed - sound not initialised\n"));
 		return false;
 	}
-	PRINTF(("snd_PlaySamplesFrom: %d At: %d Length: %d", arrayIndex, startIndex, frameCount));
-	sm_addblock(sound_stream, (unsigned char *)((int)arrayIndex + startIndex), BYTESFROMSAMPLES(frameCount));
+	PRINTF(("snd_PlaySamplesFrom: %d At: %d Length: %d", buf, startIndex, frameCount));
+	sm_addblock(sound_stream, (unsigned char *)((int)buf + startIndex), BYTESFROMSAMPLES(frameCount));
 	ssb_pause(sound_handle, false);
 	return frameCount;
 }

--- a/platforms/unix/plugins/DropPlugin/sqUnixDragDrop.c
+++ b/platforms/unix/plugins/DropPlugin/sqUnixDragDrop.c
@@ -134,6 +134,12 @@ dropRequestURI(sqInt dropIndex)	// in st coordinates
 	return uxDropFileNames[dropIndex - 1];
 }
 
+static usqIntptr_t
+fileRecordSize(void)
+{
+	return sizeof(SQFile);
+}
+
 sqInt
 dropRequestFileHandle(sqInt dropIndex)
 {

--- a/platforms/unix/vm-sound-ALSA/sqUnixSoundALSA.c
+++ b/platforms/unix/vm-sound-ALSA/sqUnixSoundALSA.c
@@ -278,13 +278,13 @@ static sqInt sound_AvailableSpace(void)
 #endif
 }
 
-static sqInt  sound_InsertSamplesFromLeadTime(sqInt frameCount, sqInt srcBufPtr, sqInt samplesOfLeadTime)	FAIL(frameCount)
+static sqInt  sound_InsertSamplesFromLeadTime(sqInt frameCount, void *srcBufPtr, sqInt samplesOfLeadTime)	FAIL(frameCount)
 
-static sqInt  sound_PlaySamplesFromAtLength(sqInt frameCount, sqInt arrayIndex, sqInt startIndex)
+static sqInt  sound_PlaySamplesFromAtLength(sqInt frameCount, void *buf, sqInt startIndex)
 {
   if (playback_handle)
     {
-      void *samples= (void *)arrayIndex + startIndex * output_channels * 2;
+      void *samples= buf + startIndex * output_channels * 2;
       int   count=   snd_pcm_writei(playback_handle, samples, frameCount);
       if (count < 0)
 	{
@@ -382,13 +382,13 @@ static double sound_GetRecordingSampleRate(void)
 # include <sys/time.h>
 
 static sqInt
-protected_sound_RecordSamplesIntoAtLength(sqInt buf, sqInt startSliceIndex, sqInt bufferSizeInBytes)
+protected_sound_RecordSamplesIntoAtLength(void *buf, sqInt startSliceIndex, sqInt bufferSizeInBytes)
 {
 	/* see git://android.git.kernel.org/platform/hardware/alsa_sound.git/
 	 *		AudioStreamInALSA.cpp: AudioStreamInALSA::read
 	 */
 	if (capture_handle) {
-		void *samples=    (void *)buf + (startSliceIndex * 2);
+		void *samples=    buf + (startSliceIndex * 2);
 # if 1
 		int   frameCount= ((bufferSizeInBytes / 2) - startSliceIndex) / input_channels;
 # else
@@ -424,12 +424,12 @@ protected_sound_RecordSamplesIntoAtLength(sqInt buf, sqInt startSliceIndex, sqIn
 }
 #endif /* PROTECT_RECORD_SAMPLES */
 
-static sqInt sound_RecordSamplesIntoAtLength(sqInt buf, sqInt startSliceIndex, sqInt bufferSizeInBytes)
+static sqInt sound_RecordSamplesIntoAtLength(void *buf, sqInt startSliceIndex, sqInt bufferSizeInBytes)
 {
 #if 0
   if (capture_handle)
     {
-      void *samples=    (void *)buf + (startSliceIndex * 2);
+      void *samples=    buf + (startSliceIndex * 2);
       int   frameCount= ((bufferSizeInBytes / 2) - startSliceIndex) / input_channels;
       int   count=      snd_pcm_readi(capture_handle, samples, frameCount);
       if (count < 0)
@@ -446,7 +446,7 @@ static sqInt sound_RecordSamplesIntoAtLength(sqInt buf, sqInt startSliceIndex, s
   return 0;
 #elif 1
 	if (capture_handle) {
-		void *samples=    (void *)buf + (startSliceIndex * 2);
+		void *samples=    buf + (startSliceIndex * 2);
 		int   frameCount= ((bufferSizeInBytes / 2) - startSliceIndex) / input_channels;
 		int   count;
 
@@ -474,7 +474,7 @@ static sqInt sound_RecordSamplesIntoAtLength(sqInt buf, sqInt startSliceIndex, s
 	 *		AudioStreamInALSA.cpp: AudioStreamInALSA::read
 	 */
 	if (capture_handle) {
-		void *samples=    (void *)buf + (startSliceIndex * 2);
+		void *samples=    buf + (startSliceIndex * 2);
 # if 0
 		int   frameCount= ((bufferSizeInBytes / 2) - startSliceIndex) / input_channels;
 # else
@@ -823,10 +823,9 @@ static void sound_SetVolume(double left, double right)
  * where 0 is the lowest recording level and 1000 is the maximum.  Do nothing
  * if the sound input hardware does not support changing the recording level.
  */
-static sqInt sound_SetRecordLevel(sqInt level)
+static void sound_SetRecordLevel(sqInt level)
 {
 	mixer_default_volume_get_set(CaptureSetting, 0, (double)level / 1000.0);
-	return 1;
 }
 
 /* Squeak Prim: Get the recording level to the given value in the range 0-1000,

--- a/platforms/unix/vm-sound-ALSA/sqUnixSoundALSA.c
+++ b/platforms/unix/vm-sound-ALSA/sqUnixSoundALSA.c
@@ -284,7 +284,7 @@ static sqInt  sound_PlaySamplesFromAtLength(sqInt frameCount, void *buf, sqInt s
 {
   if (playback_handle)
     {
-      void *samples= buf + startIndex * output_channels * 2;
+      void *samples= (char*)buf + startIndex * output_channels * 2;
       int   count=   snd_pcm_writei(playback_handle, samples, frameCount);
       if (count < 0)
 	{
@@ -388,7 +388,7 @@ protected_sound_RecordSamplesIntoAtLength(void *buf, sqInt startSliceIndex, sqIn
 	 *		AudioStreamInALSA.cpp: AudioStreamInALSA::read
 	 */
 	if (capture_handle) {
-		void *samples=    buf + (startSliceIndex * 2);
+		void *samples=    (char*)buf + (startSliceIndex * 2);
 # if 1
 		int   frameCount= ((bufferSizeInBytes / 2) - startSliceIndex) / input_channels;
 # else
@@ -429,7 +429,7 @@ static sqInt sound_RecordSamplesIntoAtLength(void *buf, sqInt startSliceIndex, s
 #if 0
   if (capture_handle)
     {
-      void *samples=    buf + (startSliceIndex * 2);
+      void *samples=    (char*)buf + (startSliceIndex * 2);
       int   frameCount= ((bufferSizeInBytes / 2) - startSliceIndex) / input_channels;
       int   count=      snd_pcm_readi(capture_handle, samples, frameCount);
       if (count < 0)
@@ -446,7 +446,7 @@ static sqInt sound_RecordSamplesIntoAtLength(void *buf, sqInt startSliceIndex, s
   return 0;
 #elif 1
 	if (capture_handle) {
-		void *samples=    buf + (startSliceIndex * 2);
+		void *samples=    (char*)buf + (startSliceIndex * 2);
 		int   frameCount= ((bufferSizeInBytes / 2) - startSliceIndex) / input_channels;
 		int   count;
 
@@ -474,7 +474,7 @@ static sqInt sound_RecordSamplesIntoAtLength(void *buf, sqInt startSliceIndex, s
 	 *		AudioStreamInALSA.cpp: AudioStreamInALSA::read
 	 */
 	if (capture_handle) {
-		void *samples=    buf + (startSliceIndex * 2);
+		void *samples=    (char*)buf + (startSliceIndex * 2);
 # if 0
 		int   frameCount= ((bufferSizeInBytes / 2) - startSliceIndex) / input_channels;
 # else

--- a/platforms/unix/vm-sound-MacOSX/sqUnixSoundMacOSX.c
+++ b/platforms/unix/vm-sound-MacOSX/sqUnixSoundMacOSX.c
@@ -710,7 +710,7 @@ static sqInt sound_InsertSamplesFromLeadTime(sqInt frameCount, void *srcBufPtr, 
       if ((frontFrames + backFrames) >= (frameCount / 2))
 	{
 	  mixFrames((short *)frontData, srcBufPtr, frontFrames);
-	  srcBufPtr += frontFrames * SqueakFrameSize;
+	  (char*)srcBufPtr += frontFrames * SqueakFrameSize;
 	  mixFrames((short *)backData,  srcBufPtr, backFrames);
 	  framesDone= frontFrames + backFrames;
 	}
@@ -733,7 +733,7 @@ static sqInt sound_PlaySamplesFromAtLength(sqInt frameCount, void *buf, sqInt st
       if (Buffer_free(output->buffer) >= byteCount)
 	{
 	  Buffer_write(output->buffer,
-		       buf + (startIndex * SqueakFrameSize),
+		       (char*)buf + (startIndex * SqueakFrameSize),
 		       byteCount);
 	  return frameCount;
 	}
@@ -864,7 +864,7 @@ static sqInt sound_RecordSamplesIntoAtLength(void *buf, sqInt startSliceIndex, s
 	  int    start= startSliceIndex * SqueakFrameSize / 2;
 	  UInt32 count= min(input->cvtBufSize, bufferSizeInBytes - start);
 	  if (kAudioHardwareNoError == AudioConverterFillBuffer(input->converter, bufferDataProc, input,
-								&count, buf + start))
+								&count, (char*)buf + start))
 	    return count / (SqueakFrameSize / 2) / input->channels;
 	}
       return 0;

--- a/platforms/unix/vm-sound-MacOSX/sqUnixSoundMacOSX.c
+++ b/platforms/unix/vm-sound-MacOSX/sqUnixSoundMacOSX.c
@@ -709,9 +709,9 @@ static sqInt sound_InsertSamplesFromLeadTime(sqInt frameCount, void *srcBufPtr, 
 
       if ((frontFrames + backFrames) >= (frameCount / 2))
 	{
-	  mixFrames((short *)frontData, (short *)srcBufPtr, frontFrames);
+	  mixFrames((short *)frontData, srcBufPtr, frontFrames);
 	  srcBufPtr += frontFrames * SqueakFrameSize;
-	  mixFrames((short *)backData,  (short *)srcBufPtr, backFrames);
+	  mixFrames((short *)backData,  srcBufPtr, backFrames);
 	  framesDone= frontFrames + backFrames;
 	}
       return framesDone;

--- a/platforms/unix/vm-sound-MacOSX/sqUnixSoundMacOSX.c
+++ b/platforms/unix/vm-sound-MacOSX/sqUnixSoundMacOSX.c
@@ -638,7 +638,7 @@ static void mixFrames(short *out, short *in, int nFrames)
 // Note: this is only used when the "sound quick start" preference is
 // enabled in the image.
 // 
-static sqInt sound_InsertSamplesFromLeadTime(sqInt frameCount, sqInt srcBufPtr, sqInt framesOfLeadTime)
+static sqInt sound_InsertSamplesFromLeadTime(sqInt frameCount, void *srcBufPtr, sqInt framesOfLeadTime)
 {
   Stream *s= output;
 
@@ -709,9 +709,9 @@ static sqInt sound_InsertSamplesFromLeadTime(sqInt frameCount, sqInt srcBufPtr, 
 
       if ((frontFrames + backFrames) >= (frameCount / 2))
 	{
-	  mixFrames((short *)frontData, (short *)pointerForOop(srcBufPtr), frontFrames);
+	  mixFrames((short *)frontData, (short *)srcBufPtr, frontFrames);
 	  srcBufPtr += frontFrames * SqueakFrameSize;
-	  mixFrames((short *)backData,  (short *)pointerForOop(srcBufPtr), backFrames);
+	  mixFrames((short *)backData,  (short *)srcBufPtr, backFrames);
 	  framesDone= frontFrames + backFrames;
 	}
       return framesDone;
@@ -725,7 +725,7 @@ static sqInt sound_InsertSamplesFromLeadTime(sqInt frameCount, sqInt srcBufPtr, 
 // play (exactly) frameCount of samples (and no less, since the result is
 // ignored).
 // 
-static sqInt sound_PlaySamplesFromAtLength(sqInt frameCount, sqInt arrayIndex, sqInt startIndex)
+static sqInt sound_PlaySamplesFromAtLength(sqInt frameCount, void *buf, sqInt startIndex)
 {
   if (output)
     {
@@ -733,7 +733,7 @@ static sqInt sound_PlaySamplesFromAtLength(sqInt frameCount, sqInt arrayIndex, s
       if (Buffer_free(output->buffer) >= byteCount)
 	{
 	  Buffer_write(output->buffer,
-		       pointerForOop(arrayIndex) + (startIndex * SqueakFrameSize),
+		       buf + (startIndex * SqueakFrameSize),
 		       byteCount);
 	  return frameCount;
 	}
@@ -855,7 +855,7 @@ static sqInt sound_StartRecording(sqInt samplesPerSec, sqInt stereo, sqInt semaI
 }
 
 
-static sqInt sound_RecordSamplesIntoAtLength(sqInt buf, sqInt startSliceIndex, sqInt bufferSizeInBytes)
+static sqInt sound_RecordSamplesIntoAtLength(void *buf, sqInt startSliceIndex, sqInt bufferSizeInBytes)
 {
   if (input)
     {
@@ -864,7 +864,7 @@ static sqInt sound_RecordSamplesIntoAtLength(sqInt buf, sqInt startSliceIndex, s
 	  int    start= startSliceIndex * SqueakFrameSize / 2;
 	  UInt32 count= min(input->cvtBufSize, bufferSizeInBytes - start);
 	  if (kAudioHardwareNoError == AudioConverterFillBuffer(input->converter, bufferDataProc, input,
-								&count, pointerForOop(buf) + start))
+								&count, buf + start))
 	    return count / (SqueakFrameSize / 2) / input->channels;
 	}
       return 0;
@@ -959,14 +959,14 @@ static void sound_SetVolume(double left, double right)
 
 // set recording gain, 0 <= level <= 1000
 // 
-static sqInt sound_SetRecordLevel(sqInt level)
+static void sound_SetRecordLevel(sqInt level)
 {
   extern int noSoundMixer;
 
   if (noSoundMixer)
-    return 0;
+    return;
 
-  return setVolume(1, (double)level / 1000.0L, (double)level / 1000.0L);
+  setVolume(1, (double)level / 1000.0L, (double)level / 1000.0L);
 }
 
 
@@ -1064,7 +1064,7 @@ int main()
       if (n)
 	{
 	  warble(n);
-	  sound_PlaySamplesFromAtLength(n, (int)sound, 0);
+	  sound_PlaySamplesFromAtLength(n, sound, 0);
 #        if (DEBUG)
 	  putchar('.');  fflush(stdout);
 #        endif

--- a/platforms/unix/vm-sound-NAS/sqUnixSoundNAS.c
+++ b/platforms/unix/vm-sound-NAS/sqUnixSoundNAS.c
@@ -129,7 +129,7 @@ static sqInt sound_Stop(void)
 {
   if(server != NULL) {
     aioDisable(AuServerConnectionNumber(server));
-		
+
     AuCloseServer(server);
     server = NULL;
   }
@@ -187,7 +187,7 @@ static sqInt sound_PlaySamplesFromAtLength(sqInt frameCount, void *srcBuf, sqInt
 	}
     }
 
-	
+
   DPRINTF("writing %d bytes (%d frames)\n", bytesToPlay, framesToPlay);
   AuWriteElement(server, flow, 0,
 		 bytesToPlay,
@@ -319,7 +319,7 @@ static AuDeviceID choose_nas_device(AuServer *server, int samplesPerSec, int ste
   return AuNone;
 }
 
-static sqInt sound_Start(int frameCount, int samplesPerSec, int stereo0, int semaIndex0)
+static sqInt sound_Start(sqInt frameCount, sqInt samplesPerSec, sqInt stereo0, sqInt semaIndex0)
 {
   AuElement elements[2];  /* first is a client element, second is
 			     a device output element */
@@ -375,7 +375,7 @@ static sqInt sound_Start(int frameCount, int samplesPerSec, int stereo0, int sem
 			    2*frameCount,   /* max: 2 buffers */
 			    frameCount,   /* low */
 			    0, NULL);
-	
+
   AuMakeElementExportDevice(&elements[1],
 			    0,
 			    device,
@@ -415,7 +415,7 @@ static sqInt sound_Start(int frameCount, int samplesPerSec, int stereo0, int sem
    XXX this routine is almost identical to snd_Start().  The two should
    be factored into a single function!
 */
-static sqInt sound_StartRecording(int desiredSamplesPerSec, int stereo0, int semaIndex0)
+static sqInt sound_StartRecording(sqInt desiredSamplesPerSec, sqInt stereo0, sqInt semaIndex0)
 {
   AuElement elements[2];  /* elements for the NAS flow to assemble:
    			        element 0 = physical input
@@ -481,7 +481,7 @@ static sqInt sound_StartRecording(int desiredSamplesPerSec, int stereo0, int sem
 			    1000000,  /* was AuUnlimitedSamples */
 			    1000, /* water mark: go ahead and send frequently! */
 			    0, NULL);
-	
+
 
 
   /* set up the flow with these elements */
@@ -521,8 +521,8 @@ static double sound_GetRecordingSampleRate(void)
 }
 
 
-static sqInt sound_RecordSamplesIntoAtLength(void *buf, int startSliceIndex,
-				  int bufferSizeInBytes)
+static sqInt sound_RecordSamplesIntoAtLength(void *buf, sqInt startSliceIndex,
+	sqInt bufferSizeInBytes)
 {
   int bytesToRead;
   int sliceSize= (stereo ? 4 : 2);   /* a "slice" seems to be a "frame": one sample from each channel */

--- a/platforms/unix/vm-sound-NAS/sqUnixSoundNAS.c
+++ b/platforms/unix/vm-sound-NAS/sqUnixSoundNAS.c
@@ -116,8 +116,8 @@ static sqInt sound_AvailableSpace(void)
   return bytesAvail;
 }
 
-static sqInt sound_InsertSamplesFromLeadTime(int frameCount, int srcBufPtr,
-				  int samplesOfLeadTime)
+static sqInt sound_InsertSamplesFromLeadTime(sqInt frameCount, void *srcBufPtr,
+				  sqInt samplesOfLeadTime)
 {
   /* not possible, I don't think using NAS */
   success(false);
@@ -141,15 +141,15 @@ static sqInt sound_Stop(void)
 
 
 
-static sqInt sound_PlaySamplesFromAtLength(int frameCount, int arrayIndex, int startIndex)
+static sqInt sound_PlaySamplesFromAtLength(sqInt frameCount, void *srcBuf, sqInt startIndex)
 {
   int bytesToPlay;
   int framesToPlay;
-  char *buf;   /* buffer to play from; it may not be arrayIndex if a
+  char *buf;   /* buffer to play from; it may not be srcBuf if a
                   conversion is necessary */
 
-  DPRINTF("PlaySamples(frameCount=%d, arrayIndex=%d, startIndex=%d\n",
-	  frameCount, arrayIndex, startIndex);
+  DPRINTF("PlaySamples(frameCount=%d, srcBuf=%d, startIndex=%d\n",
+	  frameCount, (int)srcBuf, startIndex);
 
   /* figure out how much to play */
   bytesToPlay = frameCount * bytesPerPlayFrame();
@@ -163,7 +163,7 @@ static sqInt sound_PlaySamplesFromAtLength(int frameCount, int arrayIndex, int s
      ignored */
   if(stereo)
     {
-      buf= (char *) (arrayIndex + 4*startIndex);
+      buf= (char *) (srcBuf + 4*startIndex);
     }
   else
     {
@@ -183,7 +183,7 @@ static sqInt sound_PlaySamplesFromAtLength(int frameCount, int arrayIndex, int s
 
       for(i=0; i<frameCount; i++)
 	{
-	  sbuf[i]= ((short *) (arrayIndex + 4*startIndex)) [2*i];
+	  sbuf[i]= ((short *) (srcBuf + 4*startIndex)) [2*i];
 	}
     }
 
@@ -521,7 +521,7 @@ static double sound_GetRecordingSampleRate(void)
 }
 
 
-static sqInt sound_RecordSamplesIntoAtLength(int buf, int startSliceIndex,
+static sqInt sound_RecordSamplesIntoAtLength(void *buf, int startSliceIndex,
 				  int bufferSizeInBytes)
 {
   int bytesToRead;
@@ -529,7 +529,7 @@ static sqInt sound_RecordSamplesIntoAtLength(int buf, int startSliceIndex,
 
 
   DPRINTF("RecordSamplesIntoAtLength(buf=%d, startSliceIndex=%d, bufferSizeInBytes=%d\n",
-	  buf, startSliceIndex, bufferSizeInBytes);
+	  (int)buf, startSliceIndex, bufferSizeInBytes);
 
 
   /* sanity checks */
@@ -566,9 +566,9 @@ static sqInt sound_RecordSamplesIntoAtLength(int buf, int startSliceIndex,
 
 
 /* mixer settings */
-static sqInt sound_SetRecordLevel(int level)
+static void sound_SetRecordLevel(sqInt level)
 {
-  return level;
+  return;
 }
 
 

--- a/platforms/unix/vm-sound-NAS/sqUnixSoundNAS.c
+++ b/platforms/unix/vm-sound-NAS/sqUnixSoundNAS.c
@@ -163,7 +163,7 @@ static sqInt sound_PlaySamplesFromAtLength(sqInt frameCount, void *srcBuf, sqInt
      ignored */
   if(stereo)
     {
-      buf= (char *) (srcBuf + 4*startIndex);
+      buf= (char*)srcBuf + 4*startIndex;
     }
   else
     {
@@ -183,7 +183,7 @@ static sqInt sound_PlaySamplesFromAtLength(sqInt frameCount, void *srcBuf, sqInt
 
       for(i=0; i<frameCount; i++)
 	{
-	  sbuf[i]= ((short *) (srcBuf + 4*startIndex)) [2*i];
+	  sbuf[i]= ((short *) ((char*)srcBuf + 4*startIndex)) [2*i];
 	}
     }
 
@@ -554,7 +554,7 @@ static sqInt sound_RecordSamplesIntoAtLength(void *buf, sqInt startSliceIndex,
 		flow,
 		1,     /* element 1 is the client export */
 		bytesToRead,
-		(char *) (buf + startSliceIndex*sliceSize),
+		((char*)buf + startSliceIndex*sliceSize),
 		NULL);
 
   bytesAvail -= bytesToRead;

--- a/platforms/unix/vm-sound-OSS/sqUnixSoundOSS.c
+++ b/platforms/unix/vm-sound-OSS/sqUnixSoundOSS.c
@@ -1,25 +1,25 @@
 /* sqUnixSoundOSS.c -- sound module for Open Sound System
  *
  * Author: Ian.Piumarta@squeakland.org
- * 
+ *
  * Last edited: 2008-04-21 14:51:45 by piumarta on emilia
  *
  *   Copyright (C) 1996-2005 by Ian Piumarta and other authors/contributors
  *                              listed elsewhere in this file.
  *   All rights reserved.
- *   
+ *
  *   This file is part of Unix Squeak.
- * 
+ *
  *   Permission is hereby granted, free of charge, to any person obtaining a
  *   copy of this software and associated documentation files (the "Software"),
  *   to deal in the Software without restriction, including without limitation
  *   the rights to use, copy, modify, merge, publish, distribute, sublicense,
  *   and/or sell copies of the Software, and to permit persons to whom the
  *   Software is furnished to do so, subject to the following conditions:
- * 
+ *
  *   The above copyright notice and this permission notice shall be included in
  *   all copies or substantial portions of the Software.
- * 
+ *
  *   THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  *   IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  *   FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
@@ -40,7 +40,7 @@
  * even an entire fragment) either.  Disabling the Semaphore entirely
  * increases audio performace noticably and Squeak idles for almost
  * all of the time (consuming < 1% CPU) when only playing sound.
- * 
+ *
  * If you turn the play Semaphore back on then you should at least
  * consider also turning on `soundStopWhenDone' in the image
  * preferences, otherwise Squeak WILL eat ALL of your CPU from the
@@ -342,7 +342,7 @@ static void dspClose(struct dsp *dsp)
  * If full-duplex is not supported (or if the current sample format is
  * incompatible) then stop recording before opening for playback.
  * Answer whether we successfully opened the device for output.
- * 
+ *
  * NOTE: for full-duplex operation we simply attempt to open the
  * device twice (O_RDONLY and O_WRONLY) which is NOT supported by all
  * versions of OSS [1].  If this fails we should try at least two
@@ -352,7 +352,7 @@ static void dspClose(struct dsp *dsp)
  * 2) find and open a shadow device /dev/dspN (N > 0).
  * Unfortunately I'm feeling too lazy to implement all that nonsense
  * today.
- * 
+ *
  * [1] http://www.opensound.com/readme/README.fullduplex.html
  */
 static struct dsp *dspOpen(struct dsp *dsp, int mode)
@@ -750,16 +750,16 @@ static sqInt sound_AvailableSpace(void)
 }
 
 
-static sqInt sound_InsertSamplesFromLeadTime(sqInt frameCount, sqInt srcBufPtr, sqInt samplesOfLeadTime)
+static sqInt sound_InsertSamplesFromLeadTime(sqInt frameCount, void *srcBufPtr, sqInt samplesOfLeadTime)
 {
   return success(false);
 }
 
 
-static sqInt sound_PlaySamplesFromAtLength(sqInt frameCount, sqInt arrayIndex, sqInt startIndex)
+static sqInt sound_PlaySamplesFromAtLength(sqInt frameCount, void *buf, sqInt startIndex)
 {
   assert(out->write != 0);
-  return out->write(out, pointerForOop(arrayIndex) + startIndex * out->sq.bpf, frameCount);
+  return out->write(out, buf + startIndex * out->sq.bpf, frameCount);
 }
 
 
@@ -828,7 +828,7 @@ static double sound_GetRecordingSampleRate(void)
 }
 
 
-static sqInt sound_RecordSamplesIntoAtLength(sqInt buf, sqInt startSliceIndex, sqInt bufferSizeInBytes)
+static sqInt sound_RecordSamplesIntoAtLength(void *buf, sqInt startSliceIndex, sqInt bufferSizeInBytes)
 {
   /*PRINTF(("record %d %d %d\n", buf, startSliceIndex, bufferSizeInBytes));*/
 
@@ -854,7 +854,7 @@ static sqInt sound_RecordSamplesIntoAtLength(sqInt buf, sqInt startSliceIndex, s
       frameCount= min(frameCount, framesAvail);
       /*PRINTF(("<%d", frameCount * in->hw.bpf));*/
       return in->read(in,
-		      pointerForOop(buf) + startSliceIndex * 2,
+		      buf + startSliceIndex * 2,
 		      frameCount)
 	* in->sq.channels;
     }
@@ -866,10 +866,10 @@ static sqInt sound_RecordSamplesIntoAtLength(sqInt buf, sqInt startSliceIndex, s
 
 
 /* NOTES:
- * 
+ *
  *   - output volume is connected to PCM unless there is no such
  *     device, in which case we try master VOLUME instead.
- * 
+ *
  *   - input volume is connected to RECLEVEL unless there is no such
  *     device, in which case we try IGAIN instead.
  */
@@ -1026,9 +1026,9 @@ static int sound_RecordLevel(int *level)
 #endif
 
 
-static sqInt sound_SetRecordLevel(sqInt level)
+static void sound_SetRecordLevel(sqInt level)
 {
-  if (noSoundMixer) return 1;
+  if (noSoundMixer) { success(false); return; }
   if (mixer || (mixer= mixerOpen(&dev_mixer)))
     {
       level= level * LEVEL_MAX / 1000;
@@ -1037,10 +1037,9 @@ static sqInt sound_SetRecordLevel(sqInt level)
       else if (level > 255)
 	level= 255;
 
-      if (mixerSetLevel(mixer, SOUND_MIXER_RECLEV, level, level)) return 1;
-      if (mixerSetLevel(mixer, SOUND_MIXER_IGAIN,  level, level)) return 1;
+      if (mixerSetLevel(mixer, SOUND_MIXER_RECLEV, level, level)) { success(false); return; }
+      if (mixerSetLevel(mixer, SOUND_MIXER_IGAIN,  level, level)) { success(false); return; }
     }
-  return 0;
 }
 
 static sqInt sound_SetSwitch(sqInt id, sqInt captureFlag, sqInt parameter)

--- a/platforms/unix/vm-sound-OSS/sqUnixSoundOSS.c
+++ b/platforms/unix/vm-sound-OSS/sqUnixSoundOSS.c
@@ -759,7 +759,7 @@ static sqInt sound_InsertSamplesFromLeadTime(sqInt frameCount, void *srcBufPtr, 
 static sqInt sound_PlaySamplesFromAtLength(sqInt frameCount, void *buf, sqInt startIndex)
 {
   assert(out->write != 0);
-  return out->write(out, buf + startIndex * out->sq.bpf, frameCount);
+  return out->write(out, (char*)buf + startIndex * out->sq.bpf, frameCount);
 }
 
 
@@ -854,7 +854,7 @@ static sqInt sound_RecordSamplesIntoAtLength(void *buf, sqInt startSliceIndex, s
       frameCount= min(frameCount, framesAvail);
       /*PRINTF(("<%d", frameCount * in->hw.bpf));*/
       return in->read(in,
-		      buf + startSliceIndex * 2,
+		      (char*)buf + startSliceIndex * 2,
 		      frameCount)
 	* in->sq.channels;
     }

--- a/platforms/unix/vm-sound-Sun/sqUnixSoundSun.c
+++ b/platforms/unix/vm-sound-Sun/sqUnixSoundSun.c
@@ -156,7 +156,7 @@ static int sound_AvailableSpace(void)
 
 static int sound_PlaySamplesFromAtLength(sqInt frameCount, void *srcBuf, sqInt startIndex)
 {
-  short *src= (short *) (srcBuf + 4*startIndex);
+  short *src= (short *) ((char*)srcBuf + 4*startIndex);
   short buf[2*frameCount];
   int i;
   int bytes;

--- a/platforms/unix/vm-sound-Sun/sqUnixSoundSun.c
+++ b/platforms/unix/vm-sound-Sun/sqUnixSoundSun.c
@@ -154,9 +154,9 @@ static int sound_AvailableSpace(void)
 }
 
 
-static int sound_PlaySamplesFromAtLength(int frameCount, int arrayIndex, int startIndex)
+static int sound_PlaySamplesFromAtLength(sqInt frameCount, void *srcBuf, sqInt startIndex)
 {
-  short *src= (short *) (arrayIndex + 4*startIndex);
+  short *src= (short *) (srcBuf + 4*startIndex);
   short buf[2*frameCount];
   int i;
   int bytes;
@@ -198,8 +198,8 @@ static int sound_PlaySamplesFromAtLength(int frameCount, int arrayIndex, int sta
 }
 
 
-static int sound_InsertSamplesFromLeadTime(int frameCount, int srcBufPtr,
-				  int samplesOfLeadTime)
+static int sound_InsertSamplesFromLeadTime(sqInt frameCount, void *srcBufPtr,
+				  sqInt samplesOfLeadTime)
 {
   return 0;
 }
@@ -212,10 +212,9 @@ static int sound_PlaySilence(void)
 
 
 /** recording not supported **/
-static int sound_SetRecordLevel(int level)
+static void sound_SetRecordLevel(sqInt level)
 {
   success(false);
-  return;
 }
 
 
@@ -239,7 +238,7 @@ static double sound_GetRecordingSampleRate(void)
 }
 
 
-static int sound_RecordSamplesIntoAtLength(int buf, int startSliceIndex, int bufferSizeInBytes)
+static int sound_RecordSamplesIntoAtLength(void *buf, int startSliceIndex, int bufferSizeInBytes)
 {
   success(false);
   return 0;

--- a/platforms/unix/vm-sound-Sun/sqUnixSoundSun.c
+++ b/platforms/unix/vm-sound-Sun/sqUnixSoundSun.c
@@ -3,19 +3,19 @@
  *   Copyright (C) 1996-2004 by Ian Piumarta and other authors/contributors
  *                              listed elsewhere in this file.
  *   All rights reserved.
- *   
+ *
  *   This file is part of Unix Squeak.
- * 
+ *
  *   Permission is hereby granted, free of charge, to any person obtaining a
  *   copy of this software and associated documentation files (the "Software"),
  *   to deal in the Software without restriction, including without limitation
  *   the rights to use, copy, modify, merge, publish, distribute, sublicense,
  *   and/or sell copies of the Software, and to permit persons to whom the
  *   Software is furnished to do so, subject to the following conditions:
- * 
+ *
  *   The above copyright notice and this permission notice shall be included in
  *   all copies or substantial portions of the Software.
- * 
+ *
  *   THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
  *   IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
  *   FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
@@ -25,7 +25,7 @@
  *   DEALINGS IN THE SOFTWARE.
  *
  * Authors: Ian.Piumarta@inria.fr and Lex Spoon <lex@cc.gatech.edu>
- * 
+ *
  * This support is rudimentary and is implemented largely by reading
  * header files and guessing what to do.
  */
@@ -90,13 +90,13 @@ static int sound_Stop(void)
 }
 
 
-static int sound_Start(int frameCount, int samplesPerSec, int stereo, int semaIndex)
+static sqInt sound_Start(sqInt frameCount, sqInt samplesPerSec, sqInt stereo, sqInt semaIndex)
 {
   int bytesPerFrame=	(stereo ? 4 : 2);
   int bufferBytes=	((frameCount * bytesPerFrame) / 8) * 8;
   struct audio_info info;
   int err;
-     
+
   if (auFd != -1) sound_Stop();
   auPlaySemaIndex= semaIndex;
   fmtStereo= stereo;
@@ -131,7 +131,7 @@ static int sound_Start(int frameCount, int samplesPerSec, int stereo, int semaIn
   aioEnable(auFd, 0, 0);
   aioHandle(auFd, auHandle, AIO_RX);
   return true;
-  
+
  closeAndFail:
   close(auFd);
   auFd= -1;
@@ -180,7 +180,7 @@ static int sound_PlaySamplesFromAtLength(sqInt frameCount, void *srcBuf, sqInt s
     {
       int len;
       char *pos= (char *) buf;
-	  
+
       len= write(auFd, pos, bytes);
       if (len < 0)
 	{
@@ -193,7 +193,7 @@ static int sound_PlaySamplesFromAtLength(sqInt frameCount, void *srcBuf, sqInt s
   /* add an eof marker */
   write(auFd, buf, 0);
   auBuffersPlayed += 1;
-  
+
   return frameCount;
 }
 
@@ -218,7 +218,7 @@ static void sound_SetRecordLevel(sqInt level)
 }
 
 
-static int sound_StartRecording(int desiredSamplesPerSec, int stereo, int semaIndex)
+static sqInt sound_StartRecording(sqInt desiredSamplesPerSec, sqInt stereo, sqInt semaIndex)
 {
   success(false);
   return;
@@ -238,7 +238,7 @@ static double sound_GetRecordingSampleRate(void)
 }
 
 
-static int sound_RecordSamplesIntoAtLength(void *buf, int startSliceIndex, int bufferSizeInBytes)
+static sqInt sound_RecordSamplesIntoAtLength(void *buf, sqInt startSliceIndex, sqInt bufferSizeInBytes)
 {
   success(false);
   return 0;

--- a/platforms/unix/vm-sound-custom/sqUnixCustomSound.c
+++ b/platforms/unix/vm-sound-custom/sqUnixCustomSound.c
@@ -28,13 +28,13 @@ static sqInt sound_AvailableSpace(void)
   return 8192;
 }
 
-static sqInt sound_InsertSamplesFromLeadTime(sqInt frameCount, sqInt srcBufPtr, sqInt samplesOfLeadTime)
+static sqInt sound_InsertSamplesFromLeadTime(sqInt frameCount, void *srcBufPtr, sqInt samplesOfLeadTime)
 {
   trace();
   return frameCount;
 }
 
-static sqInt sound_PlaySamplesFromAtLength(sqInt frameCount, sqInt arrayIndex, sqInt startIndex)
+static sqInt sound_PlaySamplesFromAtLength(sqInt frameCount, void *buf, sqInt startIndex)
 {
   trace();
   return frameCount;
@@ -80,7 +80,7 @@ static double sound_GetRecordingSampleRate(void)
   return 8192;
 }
 
-static sqInt sound_RecordSamplesIntoAtLength(sqInt buf, sqInt startSliceIndex, sqInt bufferSizeInBytes)
+static sqInt sound_RecordSamplesIntoAtLength(void *buf, sqInt startSliceIndex, sqInt bufferSizeInBytes)
 {
   trace();
   return 8192;
@@ -100,10 +100,9 @@ static void sound_SetVolume(double left, double right)
   trace();
 }
 
-static sqInt sound_SetRecordLevel(sqInt level)
+static void sound_SetRecordLevel(sqInt level)
 {
   trace();
-  return level;
 }
 
 

--- a/platforms/unix/vm-sound-null/sqUnixSoundNull.c
+++ b/platforms/unix/vm-sound-null/sqUnixSoundNull.c
@@ -47,7 +47,7 @@
 /* output */
 static sqInt  sound_AvailableSpace(void)									FAIL(8192)
 static sqInt  sound_InsertSamplesFromLeadTime(sqInt frameCount, void * srcBufPtr, sqInt samplesOfLeadTime)	FAIL(frameCount)
-static sqInt  sound_PlaySamplesFromAtLength(sqInt frameCount, void * arrayIndex, sqInt startIndex)		FAIL(8192)
+static sqInt  sound_PlaySamplesFromAtLength(sqInt frameCount, void * buf, sqInt startIndex)		FAIL(8192)
 static sqInt  sound_PlaySilence(void)										FAIL(8192)
 static sqInt  sound_Start(sqInt frameCount, sqInt samplesPerSec, sqInt stereo, sqInt semaIndex)			FAIL(1)
 static sqInt  sound_Stop(void)											FAIL(0)
@@ -59,7 +59,7 @@ static sqInt  sound_RecordSamplesIntoAtLength(void *buf, sqInt startSliceIndex, 
 /* mixer */
 static void   sound_Volume(double *left, double *right)								{ return; }
 static void   sound_SetVolume(double left, double right)							{ return; }
-static sqInt  sound_SetRecordLevel(sqInt level)									{ return level; }
+static void  sound_SetRecordLevel(sqInt level)									{ return; }
 
 static sqInt  sound_SetSwitch(sqInt id, sqInt captureFlag, sqInt parameter)					FAIL(-1)
 static sqInt  sound_GetSwitch(sqInt id, sqInt captureFlag, sqInt channel)					FAIL(-1)

--- a/platforms/unix/vm-sound-pulse/sqUnixSoundPulseAudio.c
+++ b/platforms/unix/vm-sound-pulse/sqUnixSoundPulseAudio.c
@@ -195,7 +195,7 @@ static void trace();
 
 static sqInt sound_AvailableSpace(void);
 static sqInt sound_InsertSamplesFromLeadTime(sqInt frameCount, void *srcBufPtr, sqInt samplesOfLeadTime);
-static sqInt sound_PlaySamplesFromAtLength(sqInt frameCount, void *arrayIndex, sqInt startIndex);
+static sqInt sound_PlaySamplesFromAtLength(sqInt frameCount, void *buf, sqInt startIndex);
 static sqInt sound_PlaySilence(void);
 static sqInt sound_Start(sqInt frameCount, sqInt samplesPerSec, sqInt stereo, sqInt semaIndex);
 static sqInt sound_Stop(void);
@@ -213,7 +213,7 @@ static int mixer_setSwitch(char *name, int captureFlag, int parameter);
 static int mixer_getSwitch(char *name, int captureFlag, int channel);
 static void sound_Volume(double *left, double *right);
 static void sound_SetVolume(double left, double right);
-static sqInt sound_SetRecordLevel(sqInt level);
+static void sound_SetRecordLevel(sqInt level);
 static sqInt sound_SetDevice(sqInt id, char *arg);
 static sqInt sound_GetSwitch(sqInt id, sqInt captureFlag, sqInt channel);
 static sqInt sound_SetSwitch(sqInt id, sqInt captureFlag, sqInt parameter);
@@ -1093,10 +1093,8 @@ static void sound_SetVolume(double left, double right) {
   trace();
 }
 
-static sqInt sound_SetRecordLevel(sqInt level) {
+static void sound_SetRecordLevel(sqInt level) {
   trace();
-  return 1;
-  return level;
 }
 
 static sqInt sound_SetDevice(sqInt id, char *arg) {

--- a/platforms/unix/vm-sound-pulse/sqUnixSoundPulseAudio.c
+++ b/platforms/unix/vm-sound-pulse/sqUnixSoundPulseAudio.c
@@ -845,7 +845,7 @@ static sqInt sound_PlaySamplesFromAtLength(sqInt frameCount, void *srcBufPtr, sq
 	
 	samples = MIN(audioOut.maxSamples, frameCount);
 	
-	if (0 == (sampleBytes = ioAddPlayBuffer(srcBufPtr + startIndex * 2 * audioOut.pa_spec.channels, samples)))
+	if (0 == (sampleBytes = ioAddPlayBuffer((char*)srcBufPtr + startIndex * 2 * audioOut.pa_spec.channels, samples)))
 		DBGMSG("sound_PlaySamplesFromAtLength(): No free buffers!");
 	
 	sigSignal(&audioOut.sigRun);
@@ -1037,7 +1037,7 @@ static sqInt sound_RecordSamplesIntoAtLength(void *buf, sqInt startSliceIndex, s
 	/*   ioGetRecordBuffer() frees the buffer after single visit. Needs more work */
 	
 	bufferNext = audioIn.bufferNext; /* preserved for debug output */
-	sampleBytes = ioGetRecordBuffer((void *)(buf + (startSliceIndex * 2)), bufferBytes);
+	sampleBytes = ioGetRecordBuffer((void *)((char*)buf + (startSliceIndex * 2)), bufferBytes);
 /*
 	if (0 < sampleBytes)
 		printf("   sound_RecordSamplesIntoAtLength(%d, %d, %d) %d, %d\n", buf, startSliceIndex, bufferSizeInBytes, bufferNext, sampleBytes);

--- a/platforms/unix/vm-sound-sndio/sqUnixSndioSound.c
+++ b/platforms/unix/vm-sound-sndio/sqUnixSndioSound.c
@@ -32,17 +32,17 @@ static sqInt sound_AvailableSpace(void)
   return  par.bufsz * par.bps * par.pchan;
 }
 
-static sqInt sound_InsertSamplesFromLeadTime(sqInt frameCount, sqInt srcBufPtr, sqInt samplesOfLeadTime)
+static sqInt sound_InsertSamplesFromLeadTime(sqInt frameCount, void *srcBufPtr, sqInt samplesOfLeadTime)
 {
   trace();
   return success(false);
 }
 
-static sqInt sound_PlaySamplesFromAtLength(sqInt frameCount, sqInt arrayIndex, sqInt startIndex)
+static sqInt sound_PlaySamplesFromAtLength(sqInt frameCount, void *buf, sqInt startIndex)
 {
   size_t bytes_played;
   trace();
-  bytes_played = sio_write(snd, pointerForOop(arrayIndex) + startIndex * (par.bps * par.pchan), frameCount * par.bps * par.pchan);
+  bytes_played = sio_write(snd, buf + startIndex * (par.bps * par.pchan), frameCount * par.bps * par.pchan);
   if (bytes_played < 0)
     return 0;
   return bytes_played / (par.bps * par.pchan);
@@ -145,7 +145,7 @@ double sound_GetRecordingSampleRate(void)
   return 8192;
 }
 
-sqInt sound_RecordSamplesIntoAtLength(sqInt buf, sqInt startSliceIndex, sqInt bufferSizeInBytes)
+sqInt sound_RecordSamplesIntoAtLength(void *buf, sqInt startSliceIndex, sqInt bufferSizeInBytes)
 {
   trace();
   return 8192;
@@ -165,10 +165,9 @@ void sound_SetVolume(double left, double right)
   trace();
 }
 
-sqInt sound_SetRecordLevel(sqInt level)
+void sound_SetRecordLevel(sqInt level)
 {
   trace();
-  return level;
 }
 
 sqInt sound_GetSwitch(sqInt id, sqInt captureFlag, sqInt channel)   { return success(true); }

--- a/platforms/unix/vm-sound-sndio/sqUnixSndioSound.c
+++ b/platforms/unix/vm-sound-sndio/sqUnixSndioSound.c
@@ -42,7 +42,7 @@ static sqInt sound_PlaySamplesFromAtLength(sqInt frameCount, void *buf, sqInt st
 {
   size_t bytes_played;
   trace();
-  bytes_played = sio_write(snd, buf + startIndex * (par.bps * par.pchan), frameCount * par.bps * par.pchan);
+  bytes_played = sio_write(snd, (char*)buf + startIndex * (par.bps * par.pchan), frameCount * par.bps * par.pchan);
   if (bytes_played < 0)
     return 0;
   return bytes_played / (par.bps * par.pchan);

--- a/platforms/unix/vm/SqSound.h
+++ b/platforms/unix/vm/SqSound.h
@@ -13,7 +13,7 @@ struct SqSound
   /* output */
   sqInt  (*snd_AvailableSpace)(void);
   sqInt  (*snd_InsertSamplesFromLeadTime)(sqInt frameCount, void *srcBufPtr, sqInt samplesOfLeadTime);
-  sqInt  (*snd_PlaySamplesFromAtLength)(sqInt frameCount, void *arrayIndex, sqInt startIndex);
+  sqInt  (*snd_PlaySamplesFromAtLength)(sqInt frameCount, void *buf, sqInt startIndex);
   sqInt  (*snd_PlaySilence)(void);
   sqInt  (*snd_Start)(sqInt frameCount, sqInt samplesPerSec, sqInt stereo, sqInt semaIndex);
   sqInt  (*snd_Stop)(void);
@@ -26,9 +26,9 @@ struct SqSound
   void   (*snd_Volume)(double *left, double *right);
   void   (*snd_SetVolume)(double left, double right);
   void   (*snd_SetRecordLevel)(sqInt level);
-  sqInt  (*snd_GetSwitch)(int id, int captureFlag, int channel);
-  sqInt  (*snd_SetSwitch)(int id, int captureFlag, int parameter);
-  sqInt  (*snd_SetDevice)(int id, char *name);
+  sqInt  (*snd_GetSwitch)(sqInt id, sqInt captureFlag, sqInt channel);
+  sqInt  (*snd_SetSwitch)(sqInt id, sqInt captureFlag, sqInt parameter);
+  sqInt  (*snd_SetDevice)(sqInt id, char *name);
 
 #if SqSoundVersionMajor > 1 || SqSoundVersionMinor >= 2
   sqInt  (*snd_SetRecordBufferFrameCount)(sqInt frameCount);

--- a/platforms/win32/plugins/SoundPlugin/sqWin32Sound.c
+++ b/platforms/win32/plugins/SoundPlugin/sqWin32Sound.c
@@ -903,7 +903,7 @@ dx_snd_PlaySamplesFromAtLength(sqInt frameCount, void *buf, sqInt startIndex)
   /* mix in stuff */
   { 
     DWORD i;
-    short *shortSrc = (short*)(buf+startIndex);
+    short *shortSrc = (short*)((char*)buf+startIndex);
     short *shortDst = (short*)dstPtr;
     dstLen /= 2;
     DSPRINTF(("|%d", dstLen));

--- a/platforms/win32/plugins/SoundPlugin/sqWin32Sound.c
+++ b/platforms/win32/plugins/SoundPlugin/sqWin32Sound.c
@@ -873,7 +873,7 @@ dx_snd_InsertSamplesFromLeadTime(sqInt frameCount, void *srcBufPtr, sqInt sample
 }
 
 static sqInt
-dx_snd_PlaySamplesFromAtLength(sqInt frameCount, void *arrayIndex, sqInt startIndex)
+dx_snd_PlaySamplesFromAtLength(sqInt frameCount, void *buf, sqInt startIndex)
 {
   HRESULT hRes;
   int bytesWritten;
@@ -903,7 +903,7 @@ dx_snd_PlaySamplesFromAtLength(sqInt frameCount, void *arrayIndex, sqInt startIn
   /* mix in stuff */
   { 
     DWORD i;
-    short *shortSrc = (short*)((char *)arrayIndex+startIndex);
+    short *shortSrc = (short*)(buf+startIndex);
     short *shortDst = (short*)dstPtr;
     dstLen /= 2;
     DSPRINTF(("|%d", dstLen));
@@ -1369,8 +1369,8 @@ snd_InsertSamplesFromLeadTime(sqInt frameCount, void *srcBufPtr, sqInt samplesOf
 }
 
 sqInt
-snd_PlaySamplesFromAtLength(sqInt frameCount, void *arrayIndex, sqInt startIndex) {
-    return dx_snd_PlaySamplesFromAtLength(frameCount, arrayIndex, startIndex);
+snd_PlaySamplesFromAtLength(sqInt frameCount, void *buf, sqInt startIndex) {
+    return dx_snd_PlaySamplesFromAtLength(frameCount, buf, startIndex);
 }
 
 sqInt

--- a/platforms/win32/vm/sqWin32Stubs.c
+++ b/platforms/win32/vm/sqWin32Stubs.c
@@ -14,16 +14,16 @@ sqInt ioDisablePowerManager(sqInt disableIfNonZero) { return success(false); }
 
 #ifdef NO_SOUND
 int snd_AvailableSpace(void) { return success(false); }
-int snd_InsertSamplesFromLeadTime(int frameCount, int srcBufPtr, int samplesOfLeadTime) { return success(false); }
-int snd_PlaySamplesFromAtLength(int frameCount, int arrayIndex, int startIndex) { return success(false); }
+int snd_InsertSamplesFromLeadTime(sqInt frameCount, void *srcBufPtr, sqInt samplesOfLeadTime) { return success(false); }
+int snd_PlaySamplesFromAtLength(sqInt frameCount, void *buf, sqInt startIndex) { return success(false); }
 int snd_PlaySilence(void) { return success(false); }
 int snd_Start(int frameCount, int samplesPerSec, int stereo, int semaIndex) { return success(false); }
 int snd_Stop(void) { return 1; }
-int snd_SetRecordLevel(int level) { return success(false); }
+void snd_SetRecordLevel(sqInt level) { success(false); }
 int snd_StartRecording(int desiredSamplesPerSec, int stereo, int semaIndex) { return success(false); }
 int snd_StopRecording(void) { return success(false); }
 double snd_GetRecordingSampleRate(void) { return success(false); }
-int snd_RecordSamplesIntoAtLength(int buf, int startSliceIndex, int bufferSizeInBytes) { return success(false); }
+int snd_RecordSamplesIntoAtLength(void *buf, int startSliceIndex, int bufferSizeInBytes) { return success(false); }
 #endif
 
 #ifdef NO_JOYSTICK

